<h1>           The Sparks Foundation          </h1>
<h2>Data Science and Business Analytics Internship</h2>
<h1>Prediction using Supervised Machine Learning</h1>
<br><p>This is the first task regarding Supervised Machine Learning.</p>
<p>This task is to <b>build a Linear Regression Model to Predict Student score for give no.of hours of study</b>.</p>

<h4>Let's import the required libraries.</h4>

In [1]:
import numpy as np
import pandas as pd

In [3]:
df=pd.read_csv("data.csv")

<p>Let's explore the given data.</p>

In [6]:
df.dtypes

Hours     float64
Scores      int64
dtype: object

In [7]:
df.head()

Hours  Scores
0    2.5      21
1    5.1      47
2    3.2      27
3    8.5      75
4    3.5      30

<p>We can see that the required data is successfully stored in a DataFrame.</p>
<h3>Let's import Linear Regression Model from sklearn.</h3>

In [8]:
from sklearn.linear_model import LinearRegression

<p>The following cells correspond to processing values and training the model with data.</p>

In [12]:
x=df.iloc[:,0].values.reshape(-1,1)
y=df.iloc[:,1].values.reshape(-1,1)

In [13]:
model=LinearRegression()
model.fit(x,y)

LinearRegression()

<h3>The task is to find the score for a student studying 9.25 hrs/day.</h3>

In [32]:
pr=np.array([9.25]).reshape(-1,1)
result=model.predict(pr)

In [30]:
model.get_params()

{'copy_X': True, 'fit_intercept': True, 'n_jobs': None, 'normalize': False}

In [36]:
result[0][0]

92.9098547701573

<p>We can see that the above model predicts that <b>the student will gain 92.9% by studying 9.25 hrs/day.</b></p> 

<h2>By<br><br>Anantha Sai Avadhanam</h2>